### PR TITLE
Remove posibility to export to onnx

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Avoid possibility to auto-export to onnx in convert_model*

### Tickets:
 - *103726*

### CODE REVIEW

**Review by eaidova — APPROVED** [review]

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py [hunk 1]
```diff
@@ -171,7 +175,7 @@ def create_pytorch_nn_module_sample_input_int32(tmp_dir):
     ref_model = make_ref_pt_model_one_input(shape, dtype=numpy.int32)
 
     return pt_model, ref_model, {'example_input': sample_input,
-                                 'input_shape': shape, "use_legacy_frontend": True}
+                                 'input': (shape, np.int32)}
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py:178]: do we have any OOB test case for PT models? Will it work now?

Comment (mvafin) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py:178]: We need at least types for such a simple model like tested here

Comment (mvafin) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_pytorch.py:178]: I added OOB test, no `input` and no `example_input` no other parameters.

**Review by rkazants — APPROVED** [review]

File: src/frontends/pytorch/src/input_model.cpp [hunk 2]
```diff
@@ -45,9 +45,13 @@ std::vector<ov::frontend::Place::Ptr> InputModel::get_inputs() const {
     for (const auto& input_idx : m_model_decoder->inputs()) {
         auto place_it = m_name_to_place.find(std::to_string(input_idx));
         FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for input.");
-        if (input_idx != 0) {
-            res.push_back(place_it->second);
+        const auto& names = place_it->second->get_names();
+        if (input_idx == 0 && std::any_of(names.cbegin(), names.cend(), [](const std::string& n) {
+                return n.find("self") != std::string::npos;
```
[the remark this answers is not in the document]
Comment (mvafin) [reply src/frontends/pytorch/src/input_model.cpp:50]: It is unlikely, but I created ticket to address this problem in future 111053

**Pull request merged**